digdag-core/src/main/java/io/digdag/core/database/DatabaseScheduleStoreManager.java

```diff
@@ -148,23 +169,29 @@ public DatabaseScheduleControlStore(Handle handle)
         }
 
         @Override
-        public boolean updateNextScheduleTime(int schedId, ScheduleTime nextTime)
+        public void updateNextScheduleTime(int schedId, ScheduleTime nextTime)
+            throws ResourceNotFoundException
         {
             int n = dao.updateNextScheduleTime(schedId,
                     nextTime.getRunTime().getEpochSecond(),
                     nextTime.getTime().getEpochSecond());
-            return n > 0;
+            if (n <= 0) {
```

Comment:
schedules.id is serial (int). so it never happens. I simply doesn't trust my understanding of jDBI or H2 enough.